CMakeLists.txt

```diff
@@ -0,0 +1,571 @@
+    ${ARITHCODING_SRC}
+    ${DRCDEC_SRC}
+    ${MPEGTPDEC_SRC} ${MPEGTPENC_SRC}
+    ${SACDEC_SRC} ${SACENC_SRC}
+    ${SBRDEC_SRC} ${SBRENC_SRC}
+    ${PCMUTILS_SRC} ${FDK_SRC} ${SYS_SRC})
+
+if(WIN32 AND BUILD_SHARED_LIBS)
+  set(libfdk_aac_SOURCES ${libfdk_aac_SOURCES} fdk-aac.def)
+endif()
+
+## Create Library target. Actually fdk-aac is enough, but we also create
+## FDK-AAC::fdk-aac for consistence with config-file package.
+
+add_library(fdk-aac ${libfdk_aac_SOURCES})
+add_library(FDK-AAC::fdk-aac ALIAS fdk-aac)
+
+## Library target configuration
+
+### Library target includes
+target_include_directories(fdk-aac
+  PUBLIC
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/libAACdec/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/libAACenc/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/libSYS/include>
+  PRIVATE
+    libArithCoding/include
+    libDRCdec/include
+    libSACdec/include
+    libSACenc/include
+    libSBRdec/include
+    libSBRenc/include
+    libMpegTPDec/include
+    libMpegTPEnc/include
+    libFDK/include
+    libPCMutils/include)
+
+### Link math library if required
+target_link_libraries(fdk-aac PRIVATE $<$<BOOL:${HAVE_LIBM}>:m>)
+
+### Set public headers and shared library version. Version info is critical for Unix-like OSes.
+set_target_properties(fdk-aac PROPERTIES
+  PUBLIC_HEADER "${fdk_aacinclude_HEADERS}"
+  VERSION 2.0.1
+  SOVERSION 2)
+
+### Some compiler options from Makefile.am
+if(MSVC)
+  target_compile_options(fdk-aac PRIVATE /EHsc)
+else()
+  target_compile_options(fdk-aac PRIVATE -fno-exceptions -fno-rtti)
+endif()
+
+## Library installation
+
+### Note we export `fdk-aac-targets` to use with config-file package.
+install(TARGETS fdk-aac EXPORT fdk-aac-targets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT BIN
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT DEV
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT DEV
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/fdk-aac COMPONENT DEV)
+
+### Configure and install pkg-config module.
+if(FDK_AAC_INSTALL_PKGCONFIG_MODULE)
+  set(prefix ${CMAKE_INSTALL_PREFIX})
+  set(exec_prefix "\$\{prefix\}")
+  set(libdir "\$\{prefix\}/${CMAKE_INSTALL_LIBDIR}")
+  set(includedir "\$\{prefix\}/${CMAKE_INSTALL_INCLUDEDIR}")
+  set(VERSION ${PROJECT_VERSION})
+  if(BUILD_SHARED_LIBS)
+    set(LIBS_PRIVATE "-lm")
+  else()
+    set(LIBS_PUBLIC "-lm")
+  endif()
+  configure_file(fdk-aac.pc.in fdk-aac.pc @ONLY)
+  install(
+    FILES
+      ${CMAKE_CURRENT_BINARY_DIR}/fdk-aac.pc
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+endif()
+
+### Configure and install config-file package. Something like pkg-config module, but for Windows.
+if(FDK_AAC_INSTALL_CMAKE_CONFIG_MODULE)
+
+  if(WIN32 AND NOT (MINGW OR CYGWIN))
+    set(CMAKE_INSTALL_PACKAGEDIR CMake)
+  else()
+    set(CMAKE_INSTALL_PACKAGEDIR ${CMAKE_INSTALL_LIBDIR}/fdk-aac/cmake)
```

Comment:
I'm not very familiar with this kind of cmake files installed along with the library, but `<prefix>/lib/fdk-aac/cmake` seems like an unusual path to install it into? Something like `<prefix>/lib/cmake/fdk-aac` would seem much more usual.

As I don't use this kind of file for finding libs myself usually, I'm considering disabling it by default, but I guess it's rather harmless to include too...

Reply:
Also for these installed cmake config files - have you tested using them in installed form to see that they work and are picked up as expected? Can you provide an example on how they're used?

Reply:
Yep, `<prefix>/lib/cmake/fdk-aac` looks better.

Reply:
I don't remember. Will check again.

In short when you what to use some library, you need to write [find module](https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#find-modules). You must write all the logic to find headers, library itself, its dependencies. This find module may use `pkg-config` if it is found.

But writing find modules for each library is not very easy. Actually, everybody use `pkg-config` to search dependencies. This is possible with CMake, but `pkg-config` is not available on every platform CMake supports. To fix that CMake introduces [config-file packages](https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html#config-file-packages). Its purpose is very similar to `pkg-config` module, but it is written in CMake and contains properly configured library target. When you import this target with `find_package`, you can use it as usual CMake library:

```cmake
find_package(fdk-aac CONFIG)
target_link_libraries(my-app PRIVATE FDK-AAC::fdk-aac)
```

Reply:
Sure. Out of common platforms, I guess only msvc based platforms is the one where pkg-config wouldn't be used? (It's not available out of the box on macos, but if going to be building custom libraries, one kind of would expect it to be available.) So wondering if it should be enabled and installed by default or not... I guess it doesn't do much harm to include it though.

Reply:
I guess it's better to install it by default.